File: frontend/src/app/models/desconto.ts
import { Parcela } from './parcela';
import { Pagamento } from './pagamento';

export class Desconto {
  id?: number;
  parcela: Parcela | null;
  pagamento: Pagamento | null;
  valor: number;
  motivo: string;

  constructor(
    parcela: Parcela | null,
    pagamento: Pagamento | null,
    valor: number,
    motivo: string,
    id?: number
  ) {
    this.id = id;
    this.parcela = parcela;
    this.pagamento = pagamento;
    this.valor = valor;
    this.motivo = motivo;
  }
}

File: backend/src/main/java/com/sentinel/backend/entity/Desconto.java
package com.sentinel.backend.entity;

import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "descontos")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Desconto {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "parcela_id")
    private Parcela parcela;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "pagamento_id")
    private Pagamento pagamento;

    private BigDecimal valor;
    private String motivo;
}
